gips/data/cdl/cdl.py

```diff
@@ -67,29 +80,48 @@ class cdlData(Data):
     _products = {
         _cdl: {
             'description': 'Crop Data Layer',
-            'assets': [_cdl],
+            'assets': [_cdl, _cdlmkii],
             'bands': [{'name': _cdl, 'units': 'none'}],
             # presently 'startdate' & 'latency' are permitted to be unspecified
             # by DH gips.utils.get_data_variables
         }
     }
 
-    _legend = None
+    def process(self, products, overwrite=False, **kwargs):
+        for asset_type, asset in self.assets.iteritems():
+            if asset_type == _cdlmkii:  # with older cdl products, the asset is the product
+                product_files = []
+
+                with ZipFile(asset.filename, 'r') as zipfile:
+                    for member in zipfile.infolist():
+                        product_files.append(zipfile.extract(member.filename))
+
+                asset_name, _ = os.path.basename(asset.filename).split('.', 1)
+
+                for f in product_files:
+                    _, ext = os.path.basename(f).split('.', 1)
+                    os.rename(f, os.path.dirname(f) + '/' + asset_name + '.' + ext)
+
+                image = GeoImage(os.path.dirname(asset.filename) + '/' + asset_name + '.tif', True)
+                image[0].SetNoData(0)
+                image = None
 
-    @classmethod
-    def legend(cls):
+    def legend(self):
         """Open the legend file, keeping it memoized for future calls."""
-        if cls._legend is None:
-            legend_fp = os.path.join(cdlRepository.get_setting('repository'), 'CDL_Legend.csv')
-            cls._legend = [row['ClassName'].lower() for row in DictReader(open(legend_fp))]
-        return cls._legend
+        if not hasattr(self, "_legend") or self._legend is None:
```

Comment:
optional:  You can `if getattr(self, "_legend", None) is not None` to shorten it a bit:

https://docs.python.org/2/library/functions.html#getattr